tests/common/include/tests/common/path_correction.inl

```diff
@@ -490,11 +445,16 @@ TEST(path_correction, cylindrical) {
     // Decrease tolerance down to 1e-8
     crk_state._tolerance = env::rk_tolerance;
 
+    // Path length per turn
+    scalar S{2.f * getter::perp(mom) / getter::norm(env::B) *
+             constant<scalar>::pi};
+    // Constrain the step size to force frequent re-navigation for this
+    // strongly bent track
+    crk_state.template set_constraint<step::constraint::e_accuracy>(0.1f * S);
+
     // Run propagator
     p.propagate(propagation, std::tie(targeter, bound_updater, rst));
 
-    EXPECT_NEAR(crk_state.path_length(), targeter._path, env::tol);
```

Comment:
Have you checked if the propagator actually propagates the reuqired path length (`S`)?

Reply:
I am pretty sure it does not. This test is not working right now. The series of PRs that I have opened right now are all steps to fix this problem. So bear with me for a bit and please ignore the ```path_correction``` file for now. It will be fixed in the end, I promise :)